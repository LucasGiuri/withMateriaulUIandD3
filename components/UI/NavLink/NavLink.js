import Link from 'next/link';
import PropTypes from 'prop-types';
import { A } from './NavLink.styles';

const NavLink = ({children, href, as}) => (
	<Link href={href} as={as}>
		<A>
			{children}
		</A>
	</Link>
);

NavLink.propTypes = {
  children: PropTypes.node.isRequired,
  href: PropTypes.string.isRequired,
  as: PropTypes.string.isRequired
};

export default NavLink;