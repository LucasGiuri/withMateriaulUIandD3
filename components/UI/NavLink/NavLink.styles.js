import styled from 'styled-components';

export const A = styled.a`
  cursor: pointer;
  text-decoration: none;
  font-family: 'Roboto', sans-serif;
`;