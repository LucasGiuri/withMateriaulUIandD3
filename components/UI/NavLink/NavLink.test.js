import React from 'react';
import renderer from 'react-test-renderer';
import NavLink from './NavLink';

describe('NavLink', () => {
  it('NavLink basic render snapshoot', () => {
    const link = 'https://google.com';
    const children = <p>Link to google</p>;
    const component = renderer.create(<NavLink href={link} as={link}>{children}</NavLink>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
