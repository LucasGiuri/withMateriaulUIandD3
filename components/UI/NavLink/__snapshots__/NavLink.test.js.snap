// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink NavLink basic render snapshoot 1`] = `
<a
  className="NavLinkstyles__A-sc-1kqds5b-0 bgqGmb"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <p>
    Link to google
  </p>
</a>
`;
